plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/UIUtils.java

```diff
@@ -289,18 +295,63 @@ public static boolean isSingleSelection(ISelection selection) {
 	 * </ul>
 	 * @return the project selected in package-, project-explorer or navigator (in this order of prescedence).
 	 */
-	public static IProject getFirstSelectedProject() {
-		IProject project = getFirstSelectedElement(PACKAGE_EXPLORER_ID, IProject.class);
-		if (project == null) {
-			project = getFirstSelectedElement(PROJECT_EXPLORER_ID, IProject.class);
-			if (project == null) {
-				project = getFirstSelectedElement(RESOURCE_NAVIGATOR_ID, IProject.class);
-			}
+	public static IProject getFirstSelectedWorkbenchProject() {
+		IViewPart part = getVisibleProjectsView();
+		if (part == null) {
+			return null;
 		}
-		return project;
+		return getFirstSelectedElement(part.getSite().getId(), IProject.class);
+	}
 
+	/**
+	 * Returns the 1st visible part among the following ones:
+	 * <ul>
+	 * <li>package explorer</li>
+	 * <li>project explorer</li>
+	 * <li>resource navigator</li>
+	 * </ul>
+	 * @return
+	 */
+	public static IViewPart getVisibleProjectsView() {
+		List<IViewPart> parts = getVisibleViewParts(PACKAGE_EXPLORER_ID, PROJECT_EXPLORER_ID, RESOURCE_NAVIGATOR_ID);
+		if (parts.isEmpty()) {
+			return null;
+		}
+		return parts.get(0);
```

Comment:
Returned list does not keep the precedence set by the order of ids. Several views can be visible at the same time in different regions of window. Method getVisibleViewParts should sort the built list to match the order of ids.